src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -0,0 +1,515 @@
+package com.databricks.labs.overwatch
+
+import com.databricks.dbutils_v1.DBUtilsHolder.dbutils
+import com.databricks.labs.overwatch.utils._
+import com.google.gson.Gson
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions._
+import org.json.{JSONArray, JSONException, JSONObject}
+import scalaj.http.{Http, HttpOptions, HttpResponse}
+
+import java.io.File
+import java.util
+
+/**
+ * Companion object for APICallV2.
+ */
+object ApiCallV2 extends SparkSessionWrapper {
+
+  def apply(apiEnv: ApiEnv, apiName: String) = {
+    new ApiCallV2(apiEnv)
+      .setApiName(apiName)
+  }
+
+  def apply(apiEnv: ApiEnv, apiName: String, queryJsonString: String) = {
+    new ApiCallV2(apiEnv)
+      .setApiName(apiName)
+      .setQuery(queryJsonString)
+  }
+
+  def apply(apiEnv: ApiEnv, apiName: String, queryJsonArray: Array[String],tempSuccessPath:String,errorTempPath:String) = {
```

Comment:
if we externalize executeBatch as mentioned below, do we still need this apply signature? If not, let's externalize and drop this signature and the steTempLocation funcs